### ISSUE TITLE
Incremental training not working on Huggingface Trans model

### ISSUE THREAD

**Issue by jvence** [MahmoudWahdan/dialog-nlu#23]
I trained a model from scratch using:
```
python3 train_joint_trans.py --train=data/custom/train --val=data/custom/valid --save=saved_models/joint_trans_model --epochs=3 --batch=64 --cache_dir=transformers_cache_dir  --trans=deepset/bert-large-uncased-whole-word-masking-squad2 --from_pt=true
```

and then attempted to incrementally train the model using:
```
#incremental training
python3 train_joint_trans.py --train=data/hp-custom/train --val=data/hp-custom/valid --save=saved_models/joint_trans_model2 --epochs=3 --batch=64 --cache_dir=transformers_cache_dir  --trans=deepset/bert-large-uncased-whole-word-masking-squad2 --from_pt=true --model=saved_models/joint_trans_model
```

This resulted in the following error:

>   File "train_joint_trans.py", line 107, in <module>
>     epochs=epochs, batch_size=batch_size, id2label=id2label)
>   File "/Users/jv/dev/project/test/ai/tensorflow/BERT-Concierge/dialog-nlu/models/base_joint_trans.py", line 79, in fit
>     callbacks=callbacks)
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 108, in _method_wrapper
>     return method(self, *args, **kwargs)
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 1098, in fit
>     tmp_logs = train_function(iterator)
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 780, in __call__
>     result = self._call(*args, **kwds)
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow/python/eager/def_function.py", line 846, in _call
>     return self._concrete_stateful_fn._filtered_call(canon_args, canon_kwds)  # pylint: disable=protected-access
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 1848, in _filtered_call
>     cancellation_manager=cancellation_manager)
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 1924, in _call_flat
>     ctx, args, cancellation_manager=cancellation_manager))
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow/python/eager/function.py", line 550, in call
>     ctx=ctx)
>   File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow/python/eager/execute.py", line 60, in quick_execute
>     inputs, attrs, num_outputs)
> tensorflow.python.framework.errors_impl.InvalidArgumentError:  Received a label value of 40 which is outside the valid range of [0, 39).

Not sure why this is happening.

**Comment by MahmoudWahdan** [comment]
Hi @jvence did you add new label?
It seems to me that you added new labels and there is a bug in incremental training that doesn't use the saved tag/intents vectorizers.
But anyway adding new labels or tags should train the model from the beginning, since it is a different architecture.

**Comment by jvence** [comment]
Yes I did add new labels. Thank you for letting me know. BTW amazing job on the project. It's come such a long way. HF support is fantastic.

**Comment by MahmoudWahdan** [comment]
Thanks @jvence for your great encouraging words.
BTW you pointed me on a bug as described in my first comment. now it's fixed a886e6f4e1f840288eae195a7dea7b43997782a6
If you try the same experiment it will give you an exception in encoding tags or intents.

**Comment by jvence** [comment]
That was quick ;-)  Thanks for the amazing work.
